client/layout/masterbar/item.jsx

```diff
@@ -36,30 +37,43 @@ class MasterbarItem extends Component {
 		}
 	};
 
-	render() {
-		const itemClasses = classNames( 'masterbar__item', this.props.className, {
-			'is-active': this.props.isActive,
-			'has-unseen': this.props.hasUnseen,
-		} );
-
+	renderChildren() {
 		return (
-			<a
-				data-tip-target={ this.props.tipTarget }
-				href={ this.props.url }
-				onClick={ this.props.onClick }
-				title={ this.props.tooltip }
-				className={ itemClasses }
-				onTouchStart={ this.preload }
-				onMouseEnter={ this.preload }
-			>
+			<Fragment>
 				{ this.props.hasUnseen && (
 					<span className="masterbar__item-bubble" aria-label="You have unseen content" />
 				) }
 				{ !! this.props.icon && <Gridicon icon={ this.props.icon } size={ 24 } /> }
 				<span className="masterbar__item-content">{ this.props.children }</span>
-			</a>
+			</Fragment>
 		);
 	}
+
+	render() {
+		const itemClasses = classNames( 'masterbar__item', this.props.className, {
+			'is-active': this.props.isActive,
+			'has-unseen': this.props.hasUnseen,
+		} );
+
+		const attributes = {
+			'data-tip-target': this.props.tipTarget,
+			onClick: this.props.onClick,
+			title: this.props.tooltip,
+			className: itemClasses,
+			onTouchStart: this.preload,
+			onMouseEnter: this.preload,
+		};
+
+		if ( this.props.url ) {
```

Comment:
Changed to a `button` element depending on whether a `url` was available for valid HTML.

Reply:
Thanks for refactoring this!